ArduPlane/quadplane.cpp

```diff
@@ -1200,16 +1227,24 @@ void QuadPlane::control_loiter()
                                                                   get_desired_yaw_rate_cds());
 
     if (plane.control_mode == &plane.mode_qland) {
-        if (poscontrol.state < QPOS_LAND_FINAL && check_land_final()) {
-            poscontrol.state = QPOS_LAND_FINAL;
-            // cut IC engine if enabled
-            if (land_icengine_cut != 0) {
-                plane.g2.ice_control.engine_control(0, 0, 0);
+        float height_above_ground = plane.relative_ground_altitude(plane.g.rangefinder_landing);
+        if (doing_precision_landing) {
```

Comment:
why the different descent rate handling for precland to normal land?
also, it omits the ICE control in precland case